src/volue/mesh/_connection.py

```diff
@@ -403,6 +403,36 @@ def run_simulation(
             for response in self.mesh_service.RunHydroSimulation(request):
                 yield None
 
+        def run_inflow_calculation(
+            self,
+            model: str,
+            area: str,
+            water_course: str,
+            start_time: datetime,
+            end_time: datetime,
+        ) -> typing.Iterator[None]:
+            if start_time is None or end_time is None:
+                raise TypeError("start_time and end_time must both have a value")
+
+            targets = self.search_for_objects(
+                f"Model/{model}/Mesh.To_Areas/{area}",
+                f"To_HydroProduction/To_WaterCourses/@[.Name={water_course}]",
+            )
+
+            if len(targets) != 1:
```

Comment:
why should this check lie in the python code? To me it makes more sense that mesh returns a grpc error...

Reply:
`simulation` parameter requires single MeshId, the `search_for_objects` may return more than 1 result and that is completely fine depending on the context where it is used. For me the check here makes perfect sense.

Reply:
I have no issue with checks, but to me it makes more sense that the grpc call would fail/handle invalid input rather than the python library/code. As it is now, it is possible to invoke the grpc procedure outside python, what then? But I guess it's a tomato/tomato thing.

Reply:
You would still fail on the client side when trying to send multiple IDs. In proto the simulation request expects single ID. The gRPC runtime shouldn't let you assemble such incorrect message that violates the protobuf declaration.